File: 042/042.func.js
// data location
const _path = '042_data.txt';

function getData(path = _path) {
  // reading data and making it manageable
  const fs = require('fs');
  let data = fs.readFileSync(path, "utf-8");
  data = data.replace(/"/g, "").split(',');
  return data;
}

// triangle number check
const isTriangle = num => Number.isInteger((-0.5 + (0.25 + 2 * num) ** 0.5));
// alphabet order for letters
const order = char => char.charCodeAt() - 64;
// word score: split -> each letter to order -> sum of orders
const score = word => word.split('').map(order).reduce((a, b) => a + b);

function triangleWords(words = getData()) {
  // every word -> score -> is this a triangle number?
  // how many triangle number? -> length
  return words.map(score).filter(isTriangle).length;
}

// results
console.log(triangleWords());

File: 042/python/042.func.py
_path = '042_data.txt' # -> path to data file

def get_data(path: str):
  ''' Reading and getting data from a file. '''
  with open(path) as file:
    # read -> remove quotes -> split by commas
    return file.read().replace(r'"', '').split(',')

# triangle number check
istriangle = lambda index: float.is_integer(-0.5 + (0.25 + 2 * index) ** 0.5)

def search(path: str, data_getter_func):
  # get data ->
  #   -> word to alphabet score
  #   -> filter non-triangles
  #   -> length of iterable means how many triangles
  words = data_getter_func(path)
  words = map(lambda word: sum(ord(letter) - 64 for letter in word), words)
  words = filter(istriangle, words)
  return len(list(words))

# results
print(search(_path, get_data))